loader/src/sources/cdc/api.js

```diff
@@ -0,0 +1,308 @@
+    if (supplyLevel > 0) {
+      sum += supplyLevel;
+    } else {
+      sum += product.in_stock ? 1 : 0;
+    }
+  }
+  return sum;
+}
+
+function formatAvailable(products) {
+  return getAvailableCount(products) > 0 ? Available.yes : Available.no;
+}
+
+const ndcLookup = {
+  // from https://www.cdc.gov/vaccines/programs/iis/COVID-19-related-codes.html
+  // and https://www2a.cdc.gov/vaccines/iis/iisstandards/vaccines.asp?rpt=ndc
+  [normalizeNDC("00310-1222-10")]: "astra_zeneca", //use
+  [normalizeNDC("00310-1222-15")]: "astra_zeneca", //sale
+  [normalizeNDC("59267-1000-01")]: "pfizer", //use
+  [normalizeNDC("59267-1000-02")]: "pfizer", //sale
+  [normalizeNDC("59267-1000-03")]: "pfizer", //sale
+  [normalizeNDC("59676-0580-05")]: "jj", //use
+  [normalizeNDC("59676-0580-15")]: "jj", //sale
+  [normalizeNDC("80631-0100-01")]: "novavax", //sale
+  [normalizeNDC("80631-0100-10")]: "novavax", //use
+  [normalizeNDC("80777-0273-10")]: "moderna", // use
+  [normalizeNDC("80777-0273-15")]: "moderna", // use
+  [normalizeNDC("80777-0273-98")]: "moderna", //sale
+  [normalizeNDC("80777-0273-99")]: "moderna", //sale
+};
+function normalizeNDC(ndcCode) {
+  // Note: this normalization makes consistent string values, not NDCs
+  const parsed = ndcCode.match(/^(\d+)-(\d+)-(\d+)$/);
+  if (!parsed) {
+    throw new Error(`Unexpected NDC format '${ndcCode}'`);
+  }
+  return parsed
+    .slice(1)
+    .map((n) => parseInt(n, 10).toString())
+    .join("~");
+}
+
+function getProductType(product) {
+  const found = ndcLookup[normalizeNDC(product.ndc)];
+  if (!found) {
+    throw new Error(
+      `Unexpected product NDC '${product.ndc}' (${JSON.stringify(product)})`
+    );
+  }
+  return found;
+}
+
+function formatProductTypes(products) {
+  return [...new Set(products.map((p) => getProductType(p)))];
+}
+
+async function checkAvailability(handler, options) {
+  const states = options.states?.split(",").map((state) => state.trim());
+
+  if (!states || !states.length) {
+    warn("No states specified for cdcApi");
+    return [];
+  }
+
+  let results = [];
+  for (const state of states) {
+    const entriesByStoreId = {};
+    for await (const entry of queryState(state)) {
+      if (!(entry.provider_location_guid in entriesByStoreId)) {
+        entriesByStoreId[entry.provider_location_guid] = [];
+      }
+      entriesByStoreId[entry.provider_location_guid].push(entry);
+    }
+
+    const stores = Object.values(entriesByStoreId);
+    const formatted = stores
+      .map(formatStore)
+      .filter(Boolean)
+      .map(markUnexpectedCVSs)
+      .forEach((item) => handler(item, { update_location: true }));
+
+    results = results.concat(formatted);
+  }
+
+  return results;
+}
+
+// 18 locations we don't have from the official CVS API that need special treatment
+// read more here: https://github.com/usdigitalresponse/univaf/pull/312#pullrequestreview-726002380
+const unexpectedCVSs = new Set([
```

Comment:
Very minor nit: everywhere else, we treat acronyms as words in camel case, so this should be `unexpectedCvses` or maybe `unexpectedCvsIds`.